src/main/scala/com/normation/rudder/datasources/UpdateHttpDataset.scala

```diff
@@ -115,13 +130,19 @@ class GetDataset(valueCompiler: InterpolatedValueCompiler) {
       body       <- QueryHttp.QUERY(datasource.httpMethod, url, headers, httpParams, datasource.sslCheck, connectionTimeout, readTimeOut) ?~! s"Error when fetching data from ${url}"
       _          =  DataSourceTimingLogger.trace(s"[${System.currentTimeMillis - time_0} ms] node '${node.id.value}': GET ${url} // ${path}")
       json       <- body match {
-                      case Some(body) => JsonSelect.fromPath(path, body) ?~! s"Error when extracting sub-json at path ${path} from ${body}"
-                      case None => Full(Nil)
+                      case Some(body) => JsonSelect.fromPath(path, body).map(x => Some(x)) ?~! s"Error when extracting sub-json at path ${path} from ${body}"
+                      // this mean we got a 404 => choose behavior based on onMissing value
+                      case None => datasource.missingNodeBehavior match {
+                        case MissingNodeBehavior.Delete              => Full(Some(Nil))
+                        case MissingNodeBehavior.DefaultValue(value) => Full(Some(JsonAST.compactRender(value) :: Nil))
+                        case MissingNodeBehavior.NoChange            => Full(None)
+                      }
                     }
     } yield {
+      // json is an Option[value :: tails] (None meaning: don't change)
       //we only get the first element from the path.
       //if list is empty, return "" (remove property).
-      DataSource.nodeProperty(datasourceName.value, json.headOption.getOrElse(""))
+      json.map( list => DataSource.nodeProperty(datasourceName.value, list.headOption.getOrElse("")) )
```

Comment:
I had to read it thrice but it's ok :) ( In first attempt, i thought the None (NoChange) would be transformed into "" and would Delete the property but understood !!